R/animint.R

```diff
@@ -966,11 +960,32 @@ saveLayer <- function(l, d, meta){
     g$nest_order <- c(g$nest_order, "group")
   }
 
+  ## Some geoms should be split into separate groups if there are NAs.
+  if(any(is.na(g.data)) && "group" %in% names(g$aes)){
+    sp.cols <- unlist(c(chunk.cols, g$nest_order))
+    order.args <- list()
+    for(sp.col in sp.cols){
+      order.args[[sp.col]] <- g.data[[sp.col]]
+    }
+    ord <- do.call(order, order.args)
+    g.data <- g.data[ord,]
+    is.missing <- apply(is.na(g.data), 1, any)
+    diff.vec <- diff(is.missing)
+    new.group.vec <- c(FALSE, diff.vec == 1)
+    for(chunk.col in sp.cols){
+      one.col <- g.data[[chunk.col]]
+      is.diff <- c(FALSE, one.col[-1] != one.col[-length(one.col)])
+      new.group.vec[is.diff] <- TRUE
+    }
+    subgroup.vec <- cumsum(new.group.vec)
+    g.data$group <- subgroup.vec
```

Comment:
cumsum is a pretty fast way to get a unique subgroup id -- this replaces group when there is for example a line with NA which should be split into several lines